tests/nn_test.py

```diff
@@ -303,7 +304,7 @@ def initializer_record(name, initializer, dtypes, min_dims=2, max_dims=4):
 ]
 
 
-@jtu.with_config(jax_legacy_prng_key="allow")
+@jtu.with_config(jax_legacy_prng_key="allow", jax_numpy_dtype_promotion="standard")
```

Comment:
We want all tests to run with `jax_numpy_dtype_promotion='strict'` unless completely necessary. If the new test requires standard promotion, you should change that within the specific test rather than changing it for the entire test suite.

Reply:
Should I create new test case with that dtype promotion modified or is there a better way to do it?

Reply:
For this particular test i need to enable x64 and disable dtype promotion.

Reply:
Makes sense: you should use `with jax.numpy_dtype_promotion('standard')` within the single test case, and you should skip the test if X64 is not enabled, something like this:
```python
    if not config.enable_x64.value:
      self.skipTest("requires x64 mode")
```